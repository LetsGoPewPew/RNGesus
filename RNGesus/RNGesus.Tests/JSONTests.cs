using ApprovalTests;
using ApprovalTests.Reporters;
using ApprovalTests.Reporters.Windows;
using Model;
using Newtonsoft.Json;
using NUnit.Framework;
using System;
using System.ComponentModel;
using System.Text;

namespace RNGesus.Tests
{
    [UseReporter(typeof(VisualStudioReporter))]
    [TestFixture]
    public class JSONTests
    {
        private BindingList<WeightedNamedOutcome> outcomes = new BindingList<WeightedNamedOutcome>();
        private string json;

        [OneTimeSetUp]
        public void Init()
        {
            outcomes.Add(new WeightedNamedOutcome("Stian1", 1, false));
            outcomes.Add(new WeightedNamedOutcome("Stian2", 2, false));
            outcomes.Add(new WeightedNamedOutcome("Stian3", 3, false));
            outcomes.Add(new WeightedNamedOutcome("Stian4", 4, false));
            outcomes.Add(new WeightedNamedOutcome("Stian5", 5, false));
            outcomes.Add(new WeightedNamedOutcome("Stian6", 6, false));
            outcomes.Add(new WeightedNamedOutcome("Stian7", 7, false));
            outcomes.Add(new WeightedNamedOutcome("Stian8", 8, false));
            outcomes.Add(new WeightedNamedOutcome("Stian9", 9, false));
            outcomes.Add(new WeightedNamedOutcome("Stian10", 10, false));

            json = JsonConvert.SerializeObject(outcomes, Formatting.Indented);
        }
        [Test]
        public void Assert_that_serializing_collection_is_correct()
        {
            Approvals.VerifyJson(json);
        }

        [Test]
        public void Assert_that_deserializing_collection_is_correct()
        {
            BindingList<WeightedNamedOutcome> deserializedJson = JsonConvert.DeserializeObject<BindingList<WeightedNamedOutcome>>(json);

            StringBuilder result = new StringBuilder();
            foreach (WeightedNamedOutcome outcome in outcomes)
            {
                result.AppendLine("Name:" + outcome.Name + " Weight: " + outcome.Weight + " Occurances: " + outcome.Occurances + " Chance to occur:" + outcome.ChanceToOccur);
            }

            Approvals.Verify(result);
        }
    }   
}
