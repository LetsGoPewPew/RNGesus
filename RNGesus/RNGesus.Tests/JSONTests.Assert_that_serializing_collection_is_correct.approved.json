[
    
  {
        
    "Name":  "Stian1",
        
    "Weight":  1,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian2",
        
    "Weight":  2,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian3",
        
    "Weight":  3,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian4",
        
    "Weight":  4,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian5",
        
    "Weight":  5,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian6",
        
    "Weight":  6,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian7",
        
    "Weight":  7,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian8",
        
    "Weight":  8,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian9",
        
    "Weight":  9,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    },
    
  {
        
    "Name":  "Stian10",
        
    "Weight":  10,
        
    "Occurances":  0,
        
    "ChanceToOccur":  0.0
  
    }

]